server/src/main.rs

```diff
@@ -109,6 +118,43 @@ fn create_session_context() -> SessionContext {
     SessionContext::with_state(state)
 }
 
+// TODO: Move this function to the configuration/metadata component.
+/// If they do not already exist, create the tables used for model table metadata. A
+/// "model_table_metadata" table that can persist model tables is created. A "columns" table that
+/// can save the index of field columns in specific tables is also created. If the tables already
+/// exist or were successfully created, return [`Ok`], otherwise return [`rusqlite::Error`].
+fn create_model_table_metadata_tables(data_folder_path: &Path) -> Result<(), rusqlite::Error> {
+    let database_path = data_folder_path.join("metadata.sqlite3");
+    let connection = Connection::open(database_path)?;
+
+    // Create the model_table_metadata SQLite table if it does not exist.
+    connection.execute(
+        "CREATE TABLE IF NOT EXISTS model_table_metadata (
+                table_name TEXT PRIMARY KEY,
+                schema TEXT NOT NULL,
+                timestamp_column_index INTEGER NOT NULL,
+                tag_column_indices BLOB NOT NULL
+        )",
+        (),
+    )?;
+
+    // Create the model_table_columns SQLite table if it does not exist.
+    connection.execute(
+        "CREATE TABLE IF NOT EXISTS model_table_columns (
+                table_name TEXT NOT NULL,
+                column_name TEXT NOT NULL,
+                column_index INTEGER NOT NULL,
```

Comment:
Maybe document that `column_index` will only use maximum 10-bits.

Reply:
Fixed.